src/main/java/com/lacunasoftware/pkiexpress/CertificateExplorer.java

```diff
@@ -66,44 +68,54 @@ public void setValidate(boolean validate) {
 	}
 
 	// endregion
-
+	
 	// region fillContent
-
+	
 	public boolean getFillContent() {
 		return fillContent;
 	}
-
+	
 	public void setFillContent(boolean fillContent) {
 		this.fillContent = fillContent;
 	}
-
+	
 	// endregion
-
+	
 	// region fillIssuer
-
+	
 	public boolean getFillIssuer() {
 		return fillIssuer;
 	}
-
+	
 	public void setFillIssuer(boolean fillIssuer) {
 		this.fillIssuer = fillIssuer;
 	}
-
+	
 	// endregion
 
+	// region trustServiceSession
+
+	public String getTrustServiceSession() {
+		return trustServiceSession;
+	}
+
+	public void setTrustServiceSession(String trustServiceSession) {
+		this.trustServiceSession = trustServiceSession;
+	}
+	
 	public CertificateExplorerResult open() throws IOException {
-		if (certificatePath == null) {
-			throw new RuntimeException("The certificate file was not set");
+		if (certificatePath == null && trustServiceSession == null) {
+			throw new RuntimeException("The certificate file or the certificate session were not set");
 		}
-
+		
 		List<String> args = new ArrayList<String>();
 		args.add("--file");
```

Comment:
Here, the new logic should be like this:
- if the field `certificatePath` is filled, add the flag with field: `--file <certificatePath>`
- if the field `trustServiceSession` is filled, add the flag with field: `--trust-service-session <session>`
The developer should pass either a file path or a session for PKIE to open the desired certificate

TLDR: You should do this:
```java
List<String> args = new ArrayList<String>();
if  (certificatePath != null) {
   args.add("--file");
   args.add(certificatePath.toString());
}

if (trustServiceSession != null) {
   args.add("--trust-service-session"):
   args.add(trustServiceSession);
}
```